source/postprocess/visualization/darcy_velocity.cc

```diff
@@ -0,0 +1,135 @@
+/*
+  Copyright (C) 2011 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/darcy_velocity.h>
+#include <aspect/utilities.h>
+#include <aspect/melt.h>
+#include <aspect/simulator.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      DarcyVelocity<dim>::
+      DarcyVelocity ()
+        :
+        DataPostprocessorVector<dim> ("darcy_velocity",
+                                      update_values | update_quadrature_points),
+        Interface<dim>()
+      {}
+
+
+
+      template <int dim>
+      std::string
+      DarcyVelocity<dim>::
+      get_physical_units () const
+      {
+        if (this->convert_output_to_years())
+          return "m/year";
+        else
+          return "m/s";
+      }
+
+
+
+      template <int dim>
+      void
+      DarcyVelocity<dim>::
+      evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                            std::vector<Vector<double>> &computed_quantities) const
+      {
+        const Quadrature<dim> &quadrature_formula = this->introspection().quadratures.velocities;
+        const unsigned int n_q_points = quadrature_formula.size();
+        AssertThrow(this->introspection().compositional_name_exists("porosity"),
+                    ExcMessage("The 'darcy velocity' postprocessor requires a compositional "
+                               "field called porosity."));
+
+        const unsigned int porosity_idx = this->introspection().find_composition_type(CompositionalFieldDescription::porosity);
+        AssertThrow(this->get_parameters().compositional_field_methods[porosity_idx] ==
+                    Parameters<dim>::AdvectionFieldMethod::fem_darcy_field,
+                    ExcMessage("The 'darcy velocity' postprocessor requires that the compositional "
+                               "field method for 'porosity' is set to 'darcy field'."));
```

Comment:
I suppose that one could do this! I'm not sure how often one would actually do this, but I'll remove this condition so that it just requires a composition called 'porosity'